nansat/mappers/mapper_metno_hfr.py

```diff
@@ -0,0 +1,134 @@
+            GRID_PX_SIZE = 5000 # Final raster px size in meters
+        else:
+            raise WrongMapperError
+
+        srs = osr.SpatialReference()
+        srs.ImportFromProj4(proj4)
+        projection = srs.ExportToWkt()
+        # Get x grid and y grid
+        x_grd, y_grd = self.create_linear_grid(nc_dataset['x'][:], nc_dataset['y'][:], GRID_PX_SIZE)
+        raster_x_size, raster_y_size = x_grd.shape
+        # Define geotransform
+        geotransform = (x_grd.min(), GRID_PX_SIZE, 0.0, y_grd.max(), 0.0, GRID_PX_SIZE * -1)
+        # Define x and y size
+        self._init_from_dataset_params(raster_x_size, raster_y_size, geotransform, projection)
+        # If required timestamp was not specified then extract date from filename and use first time
+        if timestamp is None:
+            timestamp = self.date_from_filename(filename)
+        # Comvert time info from the dataset to the datetime
+        timestamps = num2date(nc_dataset['time'][:].data, nc_dataset['time'].units)
+        # find band id for the required timestamp
+        # Note add 1 because in gdal counting starts from 1 not from 0
+        src_timestamp_id = np.where(timestamps == timestamp)[0][0] + 1
+        # Iterate through all subdatasets and bands to the dataset
+        for subdataset in gdal_dataset.GetSubDatasets():
+            # Get name of subdataset
+            subdataset_name = subdataset[0].split(':')[2]
+            # Check if the subdataset in the accepted 3D vars list
+            if subdataset_name not in self.BAND_NAMES:
+                continue
+            gdal_subdataset = gdal.Open(subdataset[0])
+            # need to be float for the nan replasement
+            band_data = gdal_subdataset.GetRasterBand(int(src_timestamp_id)).ReadAsArray().astype('float')
+            # remove fill value (replace with nan)
+            fill_value = int(gdal_subdataset.GetMetadata_Dict()['#'.join([subdataset_name, '_FillValue'])])
+            band_data[band_data == fill_value] = np.nan
+            # Interpolate data on the regular grid
+            band_grid_data = self.band2grid((nc_dataset['x'][:], nc_dataset['y'][:]), 
+                                             band_data, (x_grd, y_grd))
+            # Create VRT ffor the regridded data
+            band_vrt = VRT.from_array(band_grid_data)
+            # Add VRT to the list of all dataset vrts
+            self.band_vrts[subdataset_name + 'VRT'] = band_vrt
+            # Add band to the dataset
+            src = {'SourceFilename': self.band_vrts[subdataset_name + 'VRT'].filename, 
+                   'SourceBand': 1}
+            # Add band specific metadata
+            dst = {'name': subdataset_name}
+            for key in gdal_subdataset.GetMetadata_Dict().keys():
+                if key.startswith(subdataset_name):
+                    clean_metadata_name = key.split('#')[1]
+                    dst[clean_metadata_name] = gdal_subdataset.GetMetadata_Dict()[key]
```

Comment:
maybe exclude 'coordinates' as it gives `lat lon` which was correct in the original file but not anymore.

Reply:
Well the lon lat is not part of the bands anyway since they are not in the list of the  `BAND_NAMES` (see line 23)

Which is explicitly checked and commented on here 

```python
  76          # Check if the subdataset in the accepted 3D vars list
  77          if subdataset_name not in self.BAND_NAMES:
  78              continue
```